### PR TITLE
Updating readme and extract tests

### PR DESCRIPTION
**Pull request by caithomas6789**
- Added the ERD to the readme
- Added more extract tests

### CODE REVIEW

File: pipeline/test_extract.py [hunk 1]
```diff
@@ -2,27 +2,76 @@
...
-    """tests loads correct weather description"""
+class TestConnectToPlantIDs:
 
-    api_plants = "https://data-eng-plants-api.herokuapp.com/plants/"
-    total_num_plants = 0
-    requests_mock.get(
-        "https://data-eng-plants-api.herokuapp.com/plants/0", status_code=200)
-    response = connect_to_plant_ids(total_num_plants, api_plants)
+    def test_connect_to_plant_ids(self, requests_mock):
+        """tests loads correct weather description"""
```
Comment (IB64) [inline pipeline/test_extract.py]: the docstrings may need adjustments

File: pipeline/extract.py [hunk 2]
```diff
@@ -70,6 +70,9 @@ def connect_to_plant_ids(total_num_plants: int, api_plants: str):
 
                 plant_data.append(wanted_data)
 
+            else:
+                raise HTTPError
```
Comment (OscarB28) [inline pipeline/extract.py:74]: good way to trigger the except to raise an APIError

File: pipeline/test_extract.py [hunk 3]
```diff
@@ -2,27 +2,93 @@
...
 
-def test_connect_to_plant_ids(requests_mock):
-    """tests loads correct weather description"""
 
-    api_plants = "https://data-eng-plants-api.herokuapp.com/plants/"
-    total_num_plants = 0
-    requests_mock.get(
-        "https://data-eng-plants-api.herokuapp.com/plants/0", status_code=200)
-    response = connect_to_plant_ids(total_num_plants, api_plants)
+class TestConnectToPlantIDs:
```
Comment (OscarB28) [inline pipeline/test_extract.py:11]: classes for tests are very good

File: pipeline/test_extract.py [hunk 4]
```diff
@@ -2,27 +2,93 @@
...
+        requests_mock.get(
+            "https://data-eng-plants-api.herokuapp.com/plants/1", status_code=400)
+        with pytest.raises(APIError) as exception:
+            connect_to_plant_ids(total_num_plants, api_plants)
+
+        assert exception.value.message == {
+            'error': True, 'message': 'URL invalid.'}
+
+
+class TestGetNumberOfPlants:
```
Comment (OscarB28) [inline pipeline/test_extract.py:57]: another class, good variety of failures too

**Review by OscarB28 — APPROVED** [review]

**Pull request merged**